cmd/bundle/summary.go

```diff
@@ -0,0 +1,76 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/flags"
+	"github.com/spf13/cobra"
+)
+
+func newSummaryCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "summary",
+		Short: "Describe the bundle resources and their deployment states",
+
+		PreRunE: ConfigureBundleWithVariables,
+
+		// This command is currently intended for the Databricks VSCode extension only
+		Hidden: true,
+	}
+
+	var forcePull bool
+	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Skip local cache and load the state from the remote workspace")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		b := bundle.Get(cmd.Context())
+
+		err := bundle.Apply(cmd.Context(), b, phases.Initialize())
+		if err != nil {
+			return err
+		}
+
+		cacheDir, err := terraform.Dir(cmd.Context(), b)
+		if err != nil {
+			return err
+		}
+		_, err = os.Stat(filepath.Join(cacheDir, terraform.TerraformStateFileName))
+		noCache := errors.Is(err, os.ErrNotExist)
+
+		if forcePull || noCache {
+			err = bundle.Apply(cmd.Context(), b, terraform.StatePull())
+			if err != nil {
+				return err
+			}
+		}
+
+		err = bundle.Apply(cmd.Context(), b, terraform.Load())
+		if err != nil {
+			return err
+		}
+
+		switch root.OutputType(cmd) {
+		case flags.OutputText:
+			return fmt.Errorf("%w, only json output is supported", errors.ErrUnsupported)
```

Comment:
👍

and what happens when you don't specify an output type? https://github.com/databricks/cli/blob/deb7e67ad5018bd661d342c1e920fd5b9e084e8c/cmd/root/io.go#L34 suggests we'd get a panic?

Reply:
If you don't provide `output` flag at all the command will output the error message we return here. The panic will only happen if the command definition does't have output flag defined at all (we do it on the `cmd/root` level)

Reply:
I see so it defaults to text? And text is not supported? sg

Reply:
Yes, the text is the default. The value can also be read from the DATABRICKS_OUTPUT_FORMAT env var